02_knight_and_princess/dana_stepina/main.cpp

```diff
@@ -0,0 +1,115 @@
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+using std::cout;
+using std::endl;
+
+using VertexId = int;
+using EdgeId = int;
+
+struct Vertex {
+  VertexId id;
+  std::vector<int> edges_id;
+};
+struct Edge {
+  EdgeId id;
+  int vertex_start;
+  int vertex_end;
+};
+
+class Graph {
+ public:
+  int count_edges = 0;
+  int count_vertices = 0;
+  void create_Graph(std::vector<std::vector<int>> graph) {
+    add_Edges(graph);
+    add_Vertices(graph);
+  }
+  std::vector<Edge> get_edges() { return graph_edges_; }
+  std::vector<Vertex> get_vertices() { return graph_vertices_; }

```

Comment:
Чем копировать, потенциально большие объекты, лучше возвращать `const` ссылку
```suggestion
  const std::vector<Edge>& get_edges() { return graph_edges_; }
  const std::vector<Vertex>& get_vertices() { return graph_vertices_; }
```